src/coreclr/jit/valuenum.h

```diff
@@ -1408,6 +1411,38 @@ class ValueNumStore
     // the function application it represents; otherwise, return "false."
     bool GetVNFunc(ValueNum vn, VNFuncApp* funcApp);
 
+    // Returns "true" iff "vn" is a function application of the form "func(op1, op2)".
+    bool IsVNBinFunc(ValueNum vn, VNFunc func, ValueNum* op1 = nullptr, ValueNum* op2 = nullptr);
+
+    // Returns "true" iff "vn" is a function application of the form "func(op, cns)"
+    // the cns can be on the left side if the function is commutative.
+    template <typename T>
+    bool IsVNBinFuncWithConst(ValueNum vn, VNFunc func, ValueNum* op, T* cns)
```

Comment:
Nit: Other functions in this class seem to write out `Binary` in full (e.g. `EvalMathFuncBinary` above).

Feel free to ignore or address as part of a separate PR.